core/src/main/java/io/github/fourlastor/game/demo/round/Turn.java

```diff
@@ -3,44 +3,47 @@
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.Input;
 import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.scenes.scene2d.Actor;
 import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Image;
 import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
 import dagger.assisted.Assisted;
 import dagger.assisted.AssistedFactory;
 import dagger.assisted.AssistedInject;
-import io.github.fourlastor.game.demo.round.ability.MeleeAttackAbility;
-import io.github.fourlastor.game.demo.round.ability.MoveAbility;
-import io.github.fourlastor.game.demo.round.ability.RangedAttackAbility;
-import io.github.fourlastor.game.demo.round.ability.TileSmashAbility;
+import io.github.fourlastor.game.demo.round.ability.Abilities;
+import io.github.fourlastor.game.demo.round.monster.MonsterAbilities;
 import io.github.fourlastor.game.demo.state.GameState;
 import io.github.fourlastor.game.demo.state.unit.Unit;
-import io.github.fourlastor.game.ui.ActorSupport;
+import io.github.fourlastor.game.demo.state.unit.UnitType;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
 
 public class Turn extends RoundState {
 
     private final UnitInRound unitInRound;
     private final StateRouter router;
-    private final MoveAbility.Factory moveFactory;
-    private final MeleeAttackAbility.Factory meleeAttackFactory;
-    private final RangedAttackAbility.Factory rangedAttackFactory;
-    private final TileSmashAbility.Factory tileSmashFactory;
+    private final MonsterAbilities.DefaultAbilities defaultAbilities;
+    private final Map<UnitType, MonsterAbilities> abilitiesMap;
+    private final TextureAtlas atlas;
+
+    private final Queue<Actor> addedImages = new LinkedList<>();
 
     private boolean acted = false;
 
     @AssistedInject
     public Turn(
             @Assisted UnitInRound unitInRound,
             StateRouter router,
-            MeleeAttackAbility.Factory meleeAttackFactory,
-            RangedAttackAbility.Factory rangedAttackFactory,
-            MoveAbility.Factory moveFactory,
-            TileSmashAbility.Factory tileSmashFactory) {
+            MonsterAbilities.DefaultAbilities defaultAbilities,
+            Map<UnitType, MonsterAbilities> abilitiesMap,
```

Comment:
abilities feels inherent to a Monster, anything wrong with having monster.abilities?

Reply:
as an interface, yep, I'd still keep the (will be many) abilities implementation out, let me move things around a bit

Reply:
[6e9cb2c](https://github.com/fourlastor-jams/into-the-wip/pull/39/commits/6e9cb2c02c26180592aefa84f1409d195ebc2d0a) ok, this should be better.

The interface `Unit.Abilities` has the actual method, and it also has the `Unit.Abilities.Defaults` class.

`MonsterAbilities` got merged with `MonsterAbilities.Bindings`, so that we have a way to expose these abilities in the map